include/irr/asset/filters/CSummedAreaTableImageFilter.h

```diff
@@ -0,0 +1,402 @@
+					{
+						auto normalizeAsUsual = [&](decodeType* entryScratchAdress)
+						{
+							for (uint8_t channel = 0; channel < currentChannelCount; ++channel)
+								entryScratchAdress[channel] /= maxDecodeValues[channel];
+						};
+
+						auto normalizeAsSigned = [&](decodeType* entryScratchAdress)
+						{
+							for (uint8_t channel = 0; channel < currentChannelCount; ++channel)
+								entryScratchAdress[channel] = (2.0 * entryScratchAdress[channel] - maxDecodeValues[channel] - minDecodeValues[channel]) / (maxDecodeValues[channel] - minDecodeValues[channel]);
+						};
+
+						auto normalizeAsUnsigned = [&](decodeType* entryScratchAdress)
+						{
+							for (uint8_t channel = 0; channel < currentChannelCount; ++channel)
+								entryScratchAdress[channel] = (entryScratchAdress[channel] - minDecodeValues[channel]) / (maxDecodeValues[channel] - minDecodeValues[channel]);
+						};
+
+						std::function<void(decodeType*)> normalize;
+						{
+							if (isNormalizedFormat)
+								if (isSignedFormat)
+									normalize = normalizeAsSigned;
+								else
+									normalize = normalizeAsUnsigned;
+							else
+								normalize = normalizeAsUsual;
+						}
+
+						core::vector3du32_SIMD localCoord;
+							for (auto& z = localCoord[2] = 0u; z < state->extent.depth; ++z)
+								for (auto& y = localCoord[1] = 0u; y < state->extent.height; ++y)
+									for (auto& x = localCoord[0] = 0u; x < state->extent.width; ++x)
+									{
+										const size_t scratchOffset = asset::IImage::SBufferCopy::getLocalByteOffset(localCoord, scratchByteStrides);
+										decodeType* entryScratchAdress = (decodeType*)(reinterpret_cast<uint8_t*>(scratchMemory) + scratchOffset);
+
+										normalize(entryScratchAdress);
+									}
+					};
+					
+					bool normalized = asset::isNormalizedFormat(inFormat);
+					if (state->normalizeImageByTotalSATValues || normalized)
+						normalizeScratch(normalized, asset::isSignedFormat(inFormat));
+
+					{
+						const auto blockDims = asset::getBlockDimensions(outFormat);
+
+						auto encode = [&](uint32_t readBlockArrayOffset, core::vectorSIMDu32 readBlockPos) -> void
+						{
+							auto newReadBlockPos = (readBlockPos - decltype(readBlockPos)(state.outOffset.x, state.outOffset.y, state.outOffset.z)); // not sure
+
+							auto localOutPos = newReadBlockPos * blockDims;
+
+							// hmm
+							// shouldn't we let a user to use current handled region?
+							//auto* outDataAdress = commonExecuteData.outData + commonExecuteData.oit->getByteOffset(localOutPos, commonExecuteData.outByteStrides);
+
+							const size_t scratchOffset = asset::IImage::SBufferCopy::getLocalByteOffset(core::vector3du32_SIMD(localOutPos.x, localOutPos.y, localOutPos.z, 0), scratchByteStrides);
+							decodeType* entryScratchAdress = (decodeType*)(reinterpret_cast<uint8_t*>(scratchMemory) + scratchOffset);
+
+							for (auto blockY = 0u; blockY < blockDims.y; blockY++)
```

Comment:
https://github.com/buildaworldnet/IrrlichtBAW/pull/497/files#r470266840

Reply:
https://github.com/buildaworldnet/IrrlichtBAW/pull/497#discussion_r470679905

Reply:
remove the block iteration for loop